### PR TITLE
Introduce advanced search and bulk permissions

### PR DESCRIPTION
**Pull request by sethvargo**
This change has two major parts: advanced searching and bulk permission management.

**Advanced searching**

Advanced searching is a new feature for user search that allows for more granular searching. It's backwards-compatible with the former search syntax, but now includes the ability to filter searches by fields. Specifically, you can now query like "name:foo" or "can:APIKeyWrite" to perform advanced queries. Advanced searching is useful for bulk permission management. For example, I can search for all users that have a particular permission, then click the checkboxes next to the names that should have said permission, then remove it in a single form.

**Bulk permission management**

Bulk permission management is a new feature that allows people with UserWrite permissions to modify other users' permissions in bulk from the users index page or a user search.

Fixes https://github.com/google/exposure-notifications-verification-server/issues/1388

**Release Note**

```release-note
Advanced searching is a new feature for user search that allows for more granular searching. It's backwards-compatible with the former search syntax, but now includes the ability to filter searches by fields. Specifically, you can now query like "name:foo" or "can:APIKeyWrite" to perform advanced queries.

Bulk permission management is a new feature that allows people with `UserWrite` permissions to modify other users' permissions in bulk in their realm.
```

/assign @whaught 
/assign @mikehelmick

### CODE REVIEW

File: cmd/server/assets/users/index.html [hunk 1]
```diff
@@ -49,45 +61,105 @@
       </div>
 
       {{if $memberships}}
-        <table class="table table-bordered table-striped table-fixed table-inner-border-only border-top mb-0">
-          <thead>
-            <tr>
-              <th scope="col">Name</th>
-              <th scope="col" width="300">Email</th>
-              {{if $canWrite}}
+        <form method="POST" id="users-form">
```
Comment (sethvargo) [inline cmd/server/assets/users/index.html:64]: https://github.com/google/exposure-notifications-verification-server/pull/1525/files?w=1 is a better diff

Comment (whaught) [inline cmd/server/assets/users/index.html:64]: I think I'll need to try this out and see it in action to really verify this.

How are we going to document this? Should `#search` above have a `?` / `advanced` link with a pop up dialog explaining the advanced query options?

/lgtm
/hold

**Comment by sethvargo** [comment]
I realized last night that this implementation missing auditing. I'll need to refactor this a bit, so hold off on reviews until re-ping.

File: cmd/server/assets/users/index.html (hunk 1 above)
Comment (sethvargo) [reply cmd/server/assets/users/index.html:133]: I added an advanced query box

**Comment by sethvargo** [comment]
@whaught @mikehelmick can you re-review this. I had to make some changes for auditing.

**Review by whaught — APPROVED** [review]

**Pull request merged**